### PR TITLE
Restructure `CondTools/SiStrip`

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The goal of this PR is a heavy refactoring of the package `CondTools/SiStrip`, as a follow-up of https://github.com/cms-sw/cmssw/pull/35999.
   * plugin files have now class declarations within the implementation `.cc` file (headers are removed);
   * a bug in the `SiStripDetInfoFileReader` usage, introduced at https://github.com/cms-sw/cmssw/pull/34285, is corrected.
   * removed `SealModules.cc` file;
   * generalized modernization of the code where possible;
   * introduced test units;

#### PR validation:

Validation relies on the (augmented) unit tests, which run successfully.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by tvami** [comment]
@cmsbuild , please abort

File: CondTools/SiStrip/plugins/SiStripApvGainReader.cc [hunk 1]
```diff
@@ -1,13 +1,49 @@
-#include "CondFormats/SiStripObjects/interface/SiStripApvGain.h"
-#include "CondFormats/DataRecord/interface/SiStripApvGainRcd.h"
-#include "CalibFormats/SiStripObjects/interface/SiStripGain.h"
-
-#include "CondTools/SiStrip/plugins/SiStripApvGainReader.h"
-
+// system include files
+//#include <memory>
```
Comment (tvami) [inline CondTools/SiStrip/plugins/SiStripApvGainReader.cc:2]: should this be removed?

File: CondTools/SiStrip/plugins/SiStripLorentzAngleReader.cc [hunk 2]
```diff
@@ -1,11 +1,37 @@
-#include "CondFormats/SiStripObjects/interface/SiStripLorentzAngle.h"
-
-#include "CondTools/SiStrip/plugins/SiStripLorentzAngleReader.h"
-
+// system include files
+//#include <memory>
```
Comment (tvami) [inline CondTools/SiStrip/plugins/SiStripLorentzAngleReader.cc:2]: Should this be removed?

File: CondTools/SiStrip/plugins/SiStripNoisesReader.cc [hunk 3]
```diff
@@ -1,11 +1,39 @@
-#include "CondTools/SiStrip/plugins/SiStripNoisesReader.h"
+// system include files
+//#include <memory>
```
Comment (tvami) [inline CondTools/SiStrip/plugins/SiStripNoisesReader.cc:2]: Same comment as above

File: CondTools/SiStrip/plugins/SiStripPedestalsReader.cc [hunk 4]
```diff
@@ -1,4 +1,32 @@
...
+
+class SiStripPedestalsReader : public edm::one::EDAnalyzer<> {
+public:
+  explicit SiStripPedestalsReader(const edm::ParameterSet&);
+  ~SiStripPedestalsReader() override = default;
+
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+private:
+  uint32_t printdebug_;
```
Comment (tvami) [inline CondTools/SiStrip/plugins/SiStripPedestalsReader.cc:27]: shouldn't this be a bool after all?

File: CondTools/SiStrip/plugins/SiStripSummaryReader.cc [hunk 5]
```diff
@@ -1,13 +1,34 @@
...
 
+class SiStripSummaryReader : public edm::one::EDAnalyzer<> {
+public:
+  explicit SiStripSummaryReader(const edm::ParameterSet&);
+  ~SiStripSummaryReader() override = default;
+
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+private:
+  uint32_t printdebug_;
```
Comment (tvami) [inline CondTools/SiStrip/plugins/SiStripSummaryReader.cc:28]: shouldn't this be a bool after all?

File: CondTools/SiStrip/plugins/SiStripThresholdBuilder.cc [hunk 6]
```diff
@@ -1,8 +1,31 @@
...
+public:
+  explicit SiStripThresholdBuilder(const edm::ParameterSet& iConfig);
+
+  ~SiStripThresholdBuilder() override = default;
+
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+private:
+  edm::FileInPath fp_;
+  uint32_t printdebug_;
```
Comment (tvami) [inline CondTools/SiStrip/plugins/SiStripThresholdBuilder.cc:26]: Same as above

File: CondTools/SiStrip/plugins/SiStripThresholdReader.cc [hunk 7]
```diff
@@ -1,6 +1,33 @@
...
+
+class SiStripThresholdReader : public edm::one::EDAnalyzer<> {
+public:
+  explicit SiStripThresholdReader(const edm::ParameterSet&);
+  ~SiStripThresholdReader() override = default;
+
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+private:
+  uint32_t printdebug_;
```
Comment (tvami) [inline CondTools/SiStrip/plugins/SiStripThresholdReader.cc:27]: shouldn't this be a bool after all?

File: CondTools/SiStrip/test/SiStripFedCablingBuilder_cfg.py [hunk 8]
```diff
@@ -3,10 +3,9 @@
 process = cms.Process("FedCablingBuilder")
 
 process.MessageLogger = cms.Service("MessageLogger",
+    threshold = cms.untracked.string('INFO'),
     debugModules = cms.untracked.vstring(''),
-    cablingBuilder = cms.untracked.PSet(
-        threshold = cms.untracked.string('INFO')
-    ),
+    #destinations = cms.untracked.vstring('cout'),
```
Comment (tvami) [inline CondTools/SiStrip/test/SiStripFedCablingBuilder_cfg.py:8]: should this be removed?

**Review by tvami — COMMENTED** [review]
Some cosmetic comments. Maybe not even relevant, I do see the debug to be 5 at one point, but then in other cases you changed it to a boolean

File: CondTools/SiStrip/plugins/SiStripPedestalsReader.cc (hunk 4 above)
Comment (mmusich) [reply CondTools/SiStrip/plugins/SiStripPedestalsReader.cc:27]: No, as you can see here it's used to limit the amount of printout.

https://github.com/cms-sw/cmssw/blob/8583b733fd4047cdd9fbeb285110bdae83c73c07/CondTools/SiStrip/plugins/SiStripPedestalsReader.cc#L19

I am dismissing your comment.

File: CondTools/SiStrip/plugins/SiStripSummaryReader.cc (hunk 5 above)
Comment (mmusich) [reply CondTools/SiStrip/plugins/SiStripSummaryReader.cc:28]: perhaps, what's the functional advantage?

File: CondTools/SiStrip/plugins/SiStripThresholdBuilder.cc (hunk 6 above)
Comment (mmusich) [reply CondTools/SiStrip/plugins/SiStripThresholdBuilder.cc:26]: No, as you can see here it's used to limit the amount of printout.

https://github.com/cms-sw/cmssw/blob/8583b733fd4047cdd9fbeb285110bdae83c73c07/CondTools/SiStrip/plugins/SiStripThresholdBuilder.cc#L36

I am dismissing your comment.

File: CondTools/SiStrip/plugins/SiStripThresholdReader.cc (hunk 7 above)
Comment (mmusich) [reply CondTools/SiStrip/plugins/SiStripThresholdReader.cc:27]: No, as you can see here it's used to limit the amount of printout.

https://github.com/cms-sw/cmssw/blob/8583b733fd4047cdd9fbeb285110bdae83c73c07/CondTools/SiStrip/plugins/SiStripThresholdReader.cc#L19

I am dismissing your comment.

File: CondTools/SiStrip/test/SiStripFedCablingBuilder_cfg.py (hunk 8 above)
Comment (mmusich) [reply CondTools/SiStrip/test/SiStripFedCablingBuilder_cfg.py:8]: one might want to redirect to cout when not running the unit test.
I am dismissing the comment.

File: CondTools/SiStrip/plugins/SiStripApvGainReader.cc (hunk 1 above)
Comment (mmusich) [reply CondTools/SiStrip/plugins/SiStripApvGainReader.cc]: done.

File: CondTools/SiStrip/plugins/SiStripLorentzAngleReader.cc (hunk 2 above)
Comment (mmusich) [reply CondTools/SiStrip/plugins/SiStripLorentzAngleReader.cc]: done.

File: CondTools/SiStrip/plugins/SiStripNoisesReader.cc (hunk 3 above)
Comment (mmusich) [reply CondTools/SiStrip/plugins/SiStripNoisesReader.cc]: done.

**Comment by mmusich** [comment]
@cmsbuild , please test with #36021

**Comment by tvami** [comment]
+db
 * Jenkins tests pass
 * PR in line with the description

**Comment by mmusich** [comment]
any further comment on this PR?
Thanks

**Comment by qliphy** [comment]
+1

**Pull request merged**